libnmstate/nm/applier.py

```diff
@@ -85,23 +86,73 @@ def prepare_edited_ifaces_configuration(ifaces_desired_state):
     return con_profiles
 
 
-def set_ifaces_admin_state(ifaces_desired_state):
+def set_ifaces_admin_state(ifaces_desired_state, con_profiles=()):
+    """
+    Control interface admin state by activating, deactivating and deleting
+    devices connection profiles.
+
+    The `down` and `absent` states are handled in the same manner (temporary)
+    by deactivating the device and deleting the connection profile.
+
+    For new virtual devices, the `up` state is handled by activating the
+    new connection profile. For existing devices, the device is activated,
+    leaving it to choose the correct profile.
```

Comment:
What is the advantage of activating the device instead of always activating the profile that we want?

Reply:
I can agree with the first point. The second point seems to be wrong because the best profile will always be the one that nmstate created - otherwise we will not reach the defined state in nmstate. There is no clear argument for the third one since it seems to be more of a gut feeling.